svea_vision/scripts/sidewalk_segmentation_nanosam.py

```diff
@@ -0,0 +1,278 @@
+#! /usr/bin/env python3
+
+import rospy
+import rospkg
+import tf2_ros
+import message_filters as mf
+from cv_bridge import CvBridge
+from tf2_sensor_msgs.tf2_sensor_msgs import do_transform_cloud
+from sensor_msgs.msg import Image, PointCloud2
+
+import os
+import time
+import cv2
+import torch
+import PIL.Image
+import numpy as np
+
+from nanosam.utils.predictor import Predictor as NanoSAMPredictor
+from nanoowl.owl_predictor import OwlPredictor as NanoOwlPredictor
+
+np.float = float  # NOTE: Temporary fix for ros_numpy issue; check #39
+import ros_numpy
+
+
+def load_param(name, value=None):
+    if value is None:
+        assert rospy.has_param(name), f'Missing parameter "{name}"'
+    return rospy.get_param(name, value)
+
+def replace_base(old, new) -> str:
+    split_last = lambda xs: (xs[:-1], xs[-1])
+    is_private = new.startswith('~')
+    is_global = new.startswith('/')
+    assert not (is_private or is_global)
+    ns, _ = split_last(old.split('/'))
+    ns += new.split('/')
+    return '/'.join(ns)
+
+
+class SidewalkSegementation:
+    
+    def __init__(self) -> None:
+        try:
```

Comment:
Same question

Reply:
This is unfinished work, and the current working version doesn't use `nanosam` model, so I will probably remove this file.